contents/docs/plugins/build/reference.md

```diff
@@ -344,10 +344,81 @@ Plugins have access to some special objects in the global scope, as well as a va
 
 ### Global
 
-- `fetch`: equivalent to [node-fetch](https://www.npmjs.com/package/node-fetch)
-- `posthog`: a limited version of `posthog-node` with only `capture` supported. 
-  - `posthog.capture` signature: `capture(event: string, properties?: Record<string, any>) => void`
-  - Plugins can pass `distinct_id` on the event properties to specify what user the event should be attributed to. If `distinct_id` is not specified, the event will be a attributed to a generic "Plugin Server" person
+#### `fetch`
+
+<blockquote class="warning-note">
+
+⚠️ Be very careful when using `fetch` to send events to a PostHog instance from `processEvent` or `onEvent`! The event captured will also be run through all the installed plugins and could potentially lead to an infinite loop of event generation.
+
+</blockquote>
+
+Equivalent to [node-fetch](https://www.npmjs.com/package/node-fetch)
+
+#### posthog
+
+The global `posthog` object gives you access to the following:
+
+##### capture
+
+<blockquote class="warning-note">
+
+⚠️ Be very careful when calling `posthog.capture` from `processEvent` or `onEvent`! The event captured will also be run through all the installed plugins and could potentially lead to an infinite loop of event generation.
+
+</blockquote>
+
+Calling `posthog.capture` allows you to capture a new event under the same project the plugin is running in.
+
```

Comment:
Add a small code snippet here that shows the method signature?

Reply:
I'm potentially being a bit pick here but I'd suggest:

1. Moving the code snippet under where `posthog.capture` is mentioned.
2. Naming the arguments as flagging them as `code`
3. Explaining a little more about the structure of the `Record` argument
4. Providing an example to drill it home as was done with `await posthog.api.get` below